backends-velox/src/main/scala/io/glutenproject/utils/ArrowAbiUtil.scala

```diff
@@ -56,10 +56,68 @@ object ArrowAbiUtil {
     }
   }
 
+  def importToSchema(allocator: BufferAllocator,
+                                       cSchema: ArrowSchema): Schema = {
+    val dictProvider = new CDataDictionaryProvider
+    val schema = Data.importSchema(allocator, cSchema, dictProvider)
+    try {
+      schema
+    } finally {
+      dictProvider.close()
+    }
+  }
+
+  def exportField(allocator: BufferAllocator, field: Field, out: ArrowSchema) {
+    val dictProvider = new CDataDictionaryProvider
+    try {
+      Data.exportField(allocator, field, dictProvider, out)
+    } finally {
+      dictProvider.close()
+    }
+  }
+
+  def exportSchema(allocator: BufferAllocator, schema: Schema, out: ArrowSchema) {
+    val dictProvider = new CDataDictionaryProvider
+    try {
+      Data.exportSchema(allocator, schema, dictProvider, out)
+    } finally {
+      dictProvider.close()
+    }
+  }
+
+  def getArrowSchemaAddress(schema: Schema): Long = {
+    val out = ArrowSchema.allocateNew(SparkMemoryUtils.contextAllocator())
+    try {
+      ArrowAbiUtil.exportSchema(SparkMemoryUtils.contextAllocator(), schema, out)
+      out.memoryAddress()
+    } finally {
+      out.close()
+    }
+  }
+
+  def exportFromArrowBatch(allocator: BufferAllocator, columnarBatch: ColumnarBatch,
+                                   cSchema: ArrowSchema, cArray: ArrowArray): Unit = {
+    val vsr = toVectorSchemaRoot(allocator, columnarBatch, null)
+    try {
+      Data.exportVectorSchemaRoot(allocator, vsr, new CDataDictionaryProvider(), cArray, cSchema)
+    } catch {
+      case e: Exception =>
+        throw new RuntimeException(
+          String.format("error exporting columnar batch with schema: %s, vectors: %s",
+            vsr.getSchema, vsr.getFieldVectors), e)
+    } finally {
+      vsr.close()
+    }
+  }
 
   def exportFromSparkColumnarBatch(allocator: BufferAllocator, columnarBatch: ColumnarBatch,
-    cSchema: ArrowSchema, cArray: ArrowArray): Unit = {
-    val vsr = toVectorSchemaRoot(columnarBatch)
+                                   cSchema: ArrowSchema, cArray: ArrowArray): Unit = {
+    exportFromSparkColumnarBatch(allocator, columnarBatch, cSchema, cArray, null)
+  }
+
+  def exportFromSparkColumnarBatch(allocator: BufferAllocator, columnarBatch: ColumnarBatch,
+    cSchema: ArrowSchema, cArray: ArrowArray, arrowBatch: ArrowRecordBatch): Unit = {
```

Comment:
The same. Seems to me that we can replace this with something like

`exportFromArrowRecordBatch(allocator: BufferAllocator, arrowBatch: ArrowRecordBatch,
    cSchema: ArrowSchema, cArray: ArrowArray)`?